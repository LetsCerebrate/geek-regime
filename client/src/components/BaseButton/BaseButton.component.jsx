import React from "react";
import classnames from "classnames";

import { defaultProps, propTypes } from "./BaseButton.props";
import styles from "./BaseButton.module.scss";

BaseButton.defaultProps = defaultProps;
BaseButton.propTypes = propTypes;

function BaseButton ({
    className,
    disabled,
    onClick,
    theme,
    title,
    width,
    ...rest
}) {
    const containerClassName = classnames(
        styles.button,
        styles[`${width}Width`],
        className,
        theme && styles[`${theme}Theme`],
        (disabled) ? styles.disabled : ""
    );

    return (
        <button
            {...rest}
            className={containerClassName}
            disabled={disabled}
            onClick={onClick}
        >
            {title}
        </button>
    );
}

export default BaseButton;
