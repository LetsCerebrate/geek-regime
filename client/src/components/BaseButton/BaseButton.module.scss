@import "styles/style.scss";

.button {
    display: flex;
    justify-content: center;
    box-sizing: border-box;
    height: 5rem;
    padding: 0 3.5rem 0 3.5rem;
    outline: none;
    user-select: none;
    font-family: $font-family--normal;
    font-size: $font-size--medium;
    transition: .1s ease;
    cursor: pointer;

    &.disabled {
        opacity: 0.5;
    }

    &.autoWidth {
        width: auto;
    }

    &.fullWidth {
        width: 100%;
    }

    &.lightTheme {
        border: 1px solid $base-button-border-color-light;
        background-color: $base-button-bg-color-light;
        color: $base-button-text-color-light;
    }

    &.darkTheme {
        border: 1px solid $base-button-border-color-dark;
        background-color: $base-button-bg-color-dark;
        color: $base-button-text-color-dark;
    }
}
