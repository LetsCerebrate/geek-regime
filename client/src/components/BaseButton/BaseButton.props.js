import PropTypes from "prop-types";

const { bool, func, oneOf, string } = PropTypes;

export const defaultProps = {
    className: "",
    disabled: false,
    theme: "light",
    width: "auto"
};

export const propTypes = {
    className: string,
    disabled: bool,
    onClick: func,
    theme: oneOf(["light", "dark"]),
    title: string.isRequired,
    width: oneOf(["auto", "full"])
};
