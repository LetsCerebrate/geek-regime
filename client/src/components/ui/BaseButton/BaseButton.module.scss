@import "styles/style.scss";

.button {
    display: flex;
    align-items: center;
    justify-content: center;
    height: 5rem;
    padding: 0 3.5rem 0 3.5rem;
    outline: none;
    user-select: none;
    font-family: $font-family--normal;
    font-size: $font-size--medium;
    transition: background-color 100ms ease;
    cursor: pointer;

    &.disabled {
        opacity: 0.5;
        cursor: default;
    }

    &.autoWidth {
        width: auto;
    }

    &.fullWidth {
        width: 100%;
    }

    &.lightTheme {
        border: 1px solid $base-button-border-color-light;
        background-color: $base-button-bg-color-light;
        color: $base-button-text-color-light;

         &:not(.disabled):hover {
            background-color: $base-button-bg-color-light--hover;
        }
    }

    &.darkTheme {
        border: 1px solid $base-button-border-color-dark;
        background-color: $base-button-bg-color-dark;
        color: $base-button-text-color-dark;

         &:not(.disabled):hover {
            background-color: $base-button-bg-color-dark--hover;
        }
    }

    &.neutralTheme {
        border: 1px solid $base-button-border-color-neutral;
        background-color: $base-button-bg-color-neutral;
        color: $base-button-text-color-neutral;

        &:not(.disabled):hover {
            background-color: $base-button-bg-color-neutral--hover;
        }
    }
}
