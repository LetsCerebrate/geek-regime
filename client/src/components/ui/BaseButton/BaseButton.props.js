import {
    bool,
    func,
    oneOf,
    string
} from "prop-types";

export const defaultProps = {
    className: "",
    disabled: false,
    theme: "light",
    type: "submit",
    width: "auto"
};

export const propTypes = {
    className: string,
    disabled: bool,
    onClick: func,
    text: string.isRequired,
    theme: oneOf(["light", "dark", "neutral"]),
    type: string,
    width: oneOf(["auto", "full"])
};
