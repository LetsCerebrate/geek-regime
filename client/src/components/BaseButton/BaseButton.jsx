import React from "react";
import classnames from "classnames";

import { defaultProps, propTypes } from "./BaseButton.props";
import styles from "./BaseButton.module.scss";

BaseButton.propTypes = propTypes;
BaseButton.defaultProps = defaultProps;

function BaseButton ({
    className,
    disabled,
    onClick,
    theme,
    title,
    width,
    ...rest
}) {
    const buttonClassName = classnames(
        styles.button,
        className,
        width && styles[`${width}Width`],
        theme && styles[`${theme}Theme`],
        (disabled) ? styles.disabled : ""
    );

    return (
        <button
            {...rest}
            className={buttonClassName}
            disabled={disabled}
            onClick={onClick}
        >
            {title}
        </button>
    );
}

export default BaseButton;
