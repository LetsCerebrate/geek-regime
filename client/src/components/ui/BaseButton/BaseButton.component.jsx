import React from "react";
import classnames from "classnames";

import { defaultProps, propTypes } from "./BaseButton.props";
import styles from "./BaseButton.module.scss";

BaseButton.defaultProps = defaultProps;
BaseButton.propTypes = propTypes;

function BaseButton ({
    className,
    disabled,
    onClick,
    text,
    theme,
    type,
    width,
    ...rest
}) {
    const containerClassName = classnames(
        styles.button,
        styles[`${width}Width`],
        className,
        theme && styles[`${theme}Theme`],
        (disabled) ? styles.disabled : ""
    );

    return (
        <button
            {...rest}
            className={containerClassName}
            disabled={disabled}
            onClick={onClick}
            type={type}
        >
            {text}
        </button>
    );
}

export default BaseButton;
