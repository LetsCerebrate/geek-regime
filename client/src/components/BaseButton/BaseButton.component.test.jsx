import React from "react";
import { mount } from "enzyme";

import BaseButton from "./BaseButton.component";

it("should render with no optional props supplied", () => {
    const wrapper = mount(<BaseButton text="Click me" />);

    expect(wrapper.text()).toBe("Click me");
    expect(wrapper.prop("className")).toBe("");
    expect(wrapper.prop("disabled")).toBe(false);
    expect(wrapper.prop("onClick")).toBe(undefined);
    expect(wrapper.prop("text")).toBe("Click me");
    expect(wrapper.prop("theme")).toBe("light");
    expect(wrapper.prop("type")).toBe("submit");
    expect(wrapper.prop("width")).toBe("auto");
});

it("should render with optional props supplied", () => {
    const onClick = jest.fn();

    const wrapper = mount(
        <BaseButton
            className="testClass"
            disabled={false}
            onClick={onClick}
            text="Click me"
            theme="dark"
            type="reset"
            width="full"
        />
    );

    expect(wrapper.text()).toBe("Click me");
    expect(wrapper.prop("className")).toBe("testClass");
    expect(wrapper.prop("disabled")).toBe(false);
    expect(wrapper.prop("onClick")).toBe(onClick);
    expect(wrapper.prop("text")).toBe("Click me");
    expect(wrapper.prop("theme")).toBe("dark");
    expect(wrapper.prop("type")).toBe("reset");
    expect(wrapper.prop("width")).toBe("full");

    wrapper.simulate("click");
    wrapper.simulate("click");

    expect(onClick).toBeCalledTimes(2);
});
