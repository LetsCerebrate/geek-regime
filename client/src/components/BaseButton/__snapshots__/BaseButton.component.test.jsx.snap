// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BaseButton 1`] = `
<button
  className="button autoWidth lightTheme"
  disabled={false}
  type="submit"
>
  Click me
</button>
`;
